catroid/src/main/java/org/catrobat/catroid/common/Constants.java

```diff
@@ -165,6 +166,17 @@
 
 	public static final String WHATSAPP_URI = "whatsapp://";
 
+	// RegistrationTask Error-Messages
+	public static final String MISSING_PASSWORD = "The password is missing.";
+	public static final String NOT_BLANK_USERNAME = "Username must not be blank";
+	public static final String NOT_BLANK_EMAIL = "email must not be blank";
+	public static final String PASSWORD_LENGTH = "Your password must have at least 6 characters.";
+	public static final String INVALID_COUNTRY = "Invalid country";
+	public static final String USERNAME_EXISTS = "This username already exists.";
+	public static final String INVALID_USERNAME_VALUE = "This value is not valid.";
+	public static final String INVALID_EMAIL_ADDRESS = "Your email seems to be invalid";
+	public static final String EMAIL_ADDRESS_EXISTS = "This email address already exists.";
```

Comment:
These strings are actually localized strings from the web api. Checking for these values should be avoided imo. They are meant to be read by the user and can easily change any time.

We are passing a language value to the register web api. I don't know if it's the device language or the app language though but the api response can be a localized by using this value. So the web api could response with a localized string.

Currently it's hard for me to say how and where to fix it. Especially with having material design in mind where all errors should be reported on every input field :thinking: 

* Should the web team just respond with localized error messages?
* And/Or should the web team just respond with an error code or/and a description of the error in plain English (for developers only)
* And/Or should Catroid also implement some checking before sending a request to the server?

I'm not going into details for each point, because this it's getting confusing really quick. However, these are the three main points which should be discussed with the web and ux team first. As you know, I'm currently rarely in Graz nowadays. So it would be great if you could do it. Also pinging @thmq because he is working on material design.

The `This value is not valid.` message is definitely something the web team should fix!

Reply:
since these strings are localized from the web api, then as i guess its hard to deal with ticket [CAT-2733](https://jira.catrob.at/browse/CAT-2733). what do you think about?

Reply:
Ahm, can you please be a little more specific?

Reply:
to show the dialog with reset password depends on a specific answer from the server which is in this case "This email address already exists." if this answer was localized then i have to deal with 54 translation of this answer to show up the resetPassword dialog!
I hope you got my point?

Reply:
Ah, I got ya! Yes, an error code (or similar) would definitely be beneficial here. Especially if the error is displayed on the `EditText` itself.